awx/main/tasks/receptor.py

```diff
@@ -403,13 +403,36 @@ def run(self):
             res = self._run_internal(receptor_ctl)
             return res
         finally:
-            # Make sure to always release the work unit if we established it
-            if self.unit_id is not None and settings.RECEPTOR_RELEASE_WORK:
-                if settings.RECPETOR_KEEP_WORK_ON_ERROR and getattr(res, 'status', 'error') == 'error':
-                    try:
-                        receptor_ctl.simple_command(f"work release {self.unit_id}")
-                    except Exception:
-                        logger.exception(f"Error releasing work unit {self.unit_id}.")
+            status = getattr(res, 'status', 'error')
+            self._receptor_release_work(receptor_ctl, status)
+
+    def _receptor_release_work(self, receptor_ctl: ReceptorControl, status: str) -> None:
```

Comment:
Should we log a message on, basically, every code path taken in this function?

Reply:
fine...